pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -335,6 +372,24 @@ func (dswp *desiredStateOfWorldPopulator) deleteProcessedPod(
 	delete(dswp.pods.processedPods, podName)
 }
 
+// Removes the stale pod error records from failedProcessedPodMessages
+func (dswp *desiredStateOfWorldPopulator) cleanupFailedProcessPodMessage(
+	stopCh <-chan struct{}) {
+
+	doCleanup := func() {
+		dswp.failedProcessPodMessages.Lock()
+		defer dswp.failedProcessPodMessages.Unlock()
+
+		for msg, lastTime := range dswp.failedProcessPodMessages.messages {
+			if time.Since(lastTime) > failureEventInterval {
```

Comment:
Thinking about when this interval becomes configurable, if it is set too high, then this could cause kubelet memory to not get cleaned up for a very long time.

Reply:
yes, but any better ideas on improve this? Maybe setting a maximum time threshold?

Reply:
Brainstorming an idea: `findAndAddNewPods()` calls `podManager.GetPods()`.   When polling, can we compare the pod event to the `podManager.GetPods()` list?  If the pod is not in that list or is terminated, then we can remove it form the event cache.  I haven't thought about if this can have race conditions.

Reply:
This method we shall consider how to remove the stale events, E.g:
1. Pod with one single volume failed, but some time later, the failure volume is successfully get  processed. In this case, we also need to check pod by `dswp.podPreviouslyProcessed(uniquePodName)`, if so, we remove the staled event.
2. Pod with several failed volumes, e.g. pod with VolumeA and VolumeB, both volumes are failed to process. If some time later, VolumeA get successfully processed, in this case, we should remove stale event for VolumeA and keep event msg for VolumeB. In this case, we still need a timemout mechanism to remove VolumeA's event msg?

Reply:
My main thought is that there are two reasons to remove events from the cache:
1. All pod's volumes got successfully processed.
2. Pod is terminated/deleted so kubelet doesn't handle it anymore